docs/design/2018-08-10-restore-dropped-table.md

```diff
@@ -0,0 +1,49 @@
+
+```
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+| JOBS                                                                                                                                                                                                                          | STATE     |
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+| ID:44, Type:drop table, State:synced, SchemaState:none, SchemaID:1, TableID:39, RowCount:0, ArgLen:0, start time: 2018-08-11 11:23:53.308 +0800 CST, Err:<nil>, ErrCount:0, SnapshotVersion:0                                 | synced    |
++-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-----------+
+```
+
+As you can see, if we can restore the table before the GC worker deletes the table data, we can restore the table completely. If the table data is deleted, we can only restore an empty table.
+
+Before we run the `ADMIN RESTORE TABLE table_id`, you must:
+* Ensure that there is no any GC tasks is running, you can figure out this by using TiDB logs and metrics.
+* Enlarge the `tikv_gc_life_time` to a sufficient value.
+
+## Compatibility
+
+It's a new command and will not lead to compatibility issues.
+
+## Implementatio
+
+1. `ADMIN RESTORE TABLE table_id` will enqueue a new DDL job to TiDB general DDL queue.
+2. Before we start to do this job, we need to check if the table name is already exists(created a new table with the same table name after you drop it). If it exists, return `ErrTableExists` error.
+3. Secondly, find out whether the delete-range of the dropped table is still in the `mysql.gc_delete_range`. If not, it means that the GC worker has cleanup the data. In this situation, we cannot restore the table successfully but return an error to the client. If it is still there, we remove the record in the `mysql.gc_delete_range` table. If we successfully remove the record, continue to step 4, otherwise we return an error to the client to indicate the command cannot execute safely.
+4. Use the previous table meta infomation of the table_id to insert the meta data into the schema meta data, like what `Meta.CreateTable` does. And set the table information state to `model.StatePublic`, then the restoration will be finished after the schema is synced by all the TiDB instances.
```

Comment:
-> Use the previous table meta **information** of the table_id to insert the meta data into the schema meta data, like what `Meta.CreateTable` does. And set the table information state to `model.StatePublic`, then the restoration will be finished after the schema is synced by all the TiDB instances.